src/dbnode/storage/namespace.go

```diff
@@ -309,6 +306,7 @@ func newDatabaseNamespace(
 	seriesOpts := NewSeriesOptionsFromOptions(opts, nopts.RetentionOptions()).
 		SetStats(series.NewStats(scope)).
 		SetColdWritesEnabled(nopts.ColdWritesEnabled())
+	seriesOpts = seriesOpts.SetNamespaceId(metadata.ID())
```

Comment:
This should be `SetNamespaceID(...)` as per Go naming semantics:
https://github.com/golang/lint/issues/124

Reply:
Instead of setting the namespace ID here and looking up schema later, why not just cache the actual schema local to the namespace and use it while you hold the namespace.RLock when a series write comes in and then setting it on the `WriteOptions`?